riscv/csrs.h

```diff
@@ -783,10 +785,19 @@ class sstateen_csr_t: public hstateen_csr_t {
   virtual bool unlogged_write(const reg_t val) noexcept override;
 };
 
+// senvcfg.sse is read_only 0 when menvcfg.sse = 0
 class senvcfg_csr_t final: public envcfg_csr_t {
  public:
-  senvcfg_csr_t(processor_t* const proc, const reg_t addr, const reg_t mask, const reg_t init);
+  senvcfg_csr_t(processor_t* const proc, const reg_t addr, const reg_t mask, const reg_t init, csr_t_p menvcfg);
+
+  reg_t read() const noexcept override {
+    return (menvcfg->read() | ~(MENVCFG_SSE)) & masked_csr_t::read();
+  }
+
```

Comment:
This seems to be missing the case where v=1, menvcfg.sse is 1 but henvcfg.sse is 0..
V=0: if menvcfg.sse is 0 then senvcfg.sse is read only zero.
V=1: if henvcfg.sse is 0 then senvcfg.sse is read only zero. (since henvcfg.sse is read only zero when menvcfg.sse is 0, only testing henvcfg.sse suffices).

Reply:
fix in latest version